service/awsconfig/v3/resource_set.go

```diff
@@ -0,0 +1,309 @@
+		}
+	}
+
+	var s3BucketObjectResource framework.Resource
+	{
+		c := s3object.Config{
+			AwsService: awsService,
+			Clients: s3object.Clients{
+				S3:  config.GuestAWSClients.S3,
+				KMS: config.GuestAWSClients.KMS,
+			},
+			CloudConfig:      cloudConfig,
+			CertWatcher:      config.CertsSearcher,
+			Logger:           config.Logger,
+			RandomKeyWatcher: config.RandomkeysSearcher,
+		}
+
+		s3BucketObjectResource, err = s3object.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var cloudformationResource framework.Resource
+	{
+		c := cloudformation.Config{
+			Clients: &adapter.Clients{
+				EC2:            config.GuestAWSClients.EC2,
+				CloudFormation: config.GuestAWSClients.CloudFormation,
+				IAM:            config.GuestAWSClients.IAM,
+				KMS:            config.GuestAWSClients.KMS,
+				ELB:            config.GuestAWSClients.ELB,
+			},
+			HostClients: &adapter.Clients{
+				EC2:            config.HostAWSClients.EC2,
+				IAM:            config.HostAWSClients.IAM,
+				CloudFormation: config.HostAWSClients.CloudFormation,
+			},
+			Logger: config.Logger,
+
+			InstallationName: config.InstallationName,
+		}
+
+		cloudformationResource, err = cloudformation.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var namespaceResource framework.Resource
+	{
+		c := namespace.Config{
+			K8sClient: config.K8sClient,
+			Logger:    config.Logger,
+		}
+
+		namespaceResource, err = namespace.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var serviceResource framework.Resource
+	{
+		c := service.Config{
+			K8sClient: config.K8sClient,
+			Logger:    config.Logger,
+		}
+
+		serviceResource, err = service.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	var endpointsResource framework.Resource
+	{
+		c := endpoints.Config{
+			Clients: endpoints.Clients{
+				EC2: config.GuestAWSClients.EC2,
+			},
+			K8sClient: config.K8sClient,
+			Logger:    config.Logger,
+		}
+
+		endpointsResource, err = endpoints.New(c)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
+
+	resources := []framework.Resource{
+		kmsKeyResource,
+		s3BucketResource,
+		s3BucketObjectResource,
+		cloudformationResource,
+		namespaceResource,
+		serviceResource,
+		endpointsResource,
+	}
+
+	// Wrap resources with retry and metrics.
+	{
+		retryWrapConfig := retryresource.DefaultWrapConfig()
+
+		retryWrapConfig.BackOffFactory = func() backoff.BackOff { return backoff.WithMaxTries(backoff.NewExponentialBackOff(), ResourceRetries) }
+		retryWrapConfig.Logger = config.Logger
+
+		resources, err = retryresource.Wrap(resources, retryWrapConfig)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+
+		metricsWrapConfig := metricsresource.DefaultWrapConfig()
+
+		metricsWrapConfig.Name = config.ProjectName
+
+		resources, err = metricsresource.Wrap(resources, metricsWrapConfig)
+		if err != nil {
+			return nil, microerror.Mask(err)
+		}
+	}
```

Comment:
Wrapping is identical to the one in kvm-operator.

Reply:
That's nice. It's just the legacy resource and wrapping with retry that causes problems.

Reply:
Good to know.